libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -1710,6 +1711,106 @@ namespace meshkernelapi
         return exitCode;
     }
 
+    MKERNEL_API int mkernel_splines_snap_to_landboundary(int meshKernelId,
+                                                         const GeometryList& land,
+                                                         GeometryList& splines,
+                                                         int startSplineIndex,
+                                                         int endSplineIndex)
+    {
+        int exitCode = Success;
+
+        try
+        {
+            if (meshKernelState.count(meshKernelId) == 0)
+            {
+                throw std::invalid_argument("MeshKernel: The selected mesh kernel id does not exist.");
+            }
+
+            if (startSplineIndex > endSplineIndex)
+            {
+                throw std::invalid_argument("Invalid spline range: " + std::to_string(startSplineIndex) + " > " + std::to_string(endSplineIndex));
+            }
+
+            if (land.num_coordinates == 0)
+            {
+                throw std::invalid_argument("Land boundary has no point values.");
+            }
+
+            if (land.coordinates_x == nullptr || land.coordinates_y == nullptr)
+            {
+                throw std::invalid_argument("Land boundary data is null.");
+            }
+
+            if (splines.num_coordinates == 0)
+            {
+                throw std::invalid_argument("Spline has no point values.");
+            }
+
+            if (splines.coordinates_x == nullptr || splines.coordinates_y == nullptr)
+            {
+                throw std::invalid_argument("Spline data is null.");
+            }
+
+            std::vector<meshkernel::Point> landBoundaryPoints(land.num_coordinates);
+            std::vector<meshkernel::Point> splinePoints(splines.num_coordinates);
+
+            //--------------------------------
+            // Copy points from parameters to create land boundary and splines
+
+            for (int i = 0; i < land.num_coordinates; ++i)
+            {
+                landBoundaryPoints[i] = meshkernel::Point({land.coordinates_x[i], land.coordinates_y[i]});
+            }
+
+            for (int i = 0; i < splines.num_coordinates; ++i)
+            {
+                splinePoints[i] = meshkernel::Point({splines.coordinates_x[i], splines.coordinates_y[i]});
+            }
+
+            meshkernel::LandBoundary landBoundary(landBoundaryPoints);
+            meshkernel::Splines splineValues(meshKernelState[meshKernelId].m_mesh2d->m_projection);
+
+            splineValues.AddSpline(splinePoints, startSplineIndex, static_cast<meshkernel::UInt>(splinePoints.size()));
```

Comment:
suppose we pass to the API only the spline that needs to be snapped in a set of several splines, the startSplineIndex and endSplineIndex would not be required. what we need instead are the indeces of the the corner points defining the portion of the spline that will be snapped